src/types/list.rs

```diff
@@ -23,13 +23,24 @@ impl PyList {
     pub fn new<'p, T, U>(
         py: Python<'p>,
         elements: impl IntoIterator<Item = T, IntoIter = U>,
+        capacity: Option<usize>,
```

Comment:
I don't think we need to add capacity arg here.
No function in Rust's stdlib takes both iterator and capacity as argument so I think it's confusing

Reply:
Do we need a third constructor in that case?

Reply:
I don't think so.
How about adding a `PyList::append_iter` and use it like
```rust
let list = PyList::with_capacity(~);
list.append_iter(iter);
```
?